### DISCUSSION TITLE
SWI-Prolog object notation

### DISCUSSION THREAD

**Discussion by vasily-kartashov** [tuProlog/2p-kt#13] [Ideas]
SWI-Prolog has dicts https://www.swi-prolog.org/pldoc/man?section=bidicts which could help to solve the problem of _transparently_ mapping entities from and to Java (I looked at the 2p-kt presentation, so that's why I stress the word _transparently_ here).

I'd like to be able to write something as simple as this:
```prolog
:- pool(db, _{ url      : "jdbc:mysql://tfb-database:3306/hello_world"
             , driver   : "com.mysql.cj.jdbc.Driver"
             , user     : "benchmarkdbuser"
             , password : "benchmarkdbpass"
             }, 512).
```

Is it possible to support the same syntax without changing the parser?

**Comment by gciatto** [comment]
Mmh no, supporting the dicts requires changes at several levels, including the parser but also the core.

It's an interesting feature, so I'll definitely have a look. However there are poor chances I'll do that in the short term.

Il you only need the square brackets notation without tag and without a fancy way to access keys by dot, you bal exploit blocks (ie terms among square brackets such as `{a, b, c}` plus a custom operator `op(800, xfx, ':')` which would let you write `{k:v, k1:v1} `.

Otherwise, have a look to the `oop-lib`, which let's you manipule java objects from Prolog. But that experimental.
